2_process/src/pair_nhd_catchments.R

```diff
@@ -0,0 +1,124 @@
+pair_nhd_catchments <- function(prms_line,prms_hrus,min_area_overlap,xwalk_table,nhd_lines){
+  #' @description This function finds the NHDPlusV2 catchments that overlap the HRU's for each PRMS segment in the DRB. 
+  #' For the majority of DRB segments, this function finds the catchments that directly drain to a given segment 
+  #' by navigating the NHDPlusV2 value-added attributes tables.  
+  #' For a subset of segments, the contributing area by NHD navigation differed considerably from the 
+  #' contributing area represented by the corresponding HRU polygons (based on visual inspection). This
+  #' function applies special handling for those select segments and uses a spatial join to 
+  #' find the NHDPlusV2 catchments that overlap the PRMS HRU's.
+  #' 
+  #' @param prms_line sf linestring containing a target PRMS segment
+  #' @param prms_hrus sf (multi)polygon containing the HRU's from the NHGF
+  #' @param min_area_overlap float; value indicating the minimum proportion of NHDPlusV2 catchment area that 
+  #' overlaps the PRMS polygon in order to be retained
+  #' @param xwalk_table data frame containing the NHDPlusV2/PRMS reach-to-segment crosswalk information
+  #' @param nhd_lines sf object containing NHDPlusV2 flowlines for area of interest
+  #' nhd_lines must contain variables COMID, PATHLENGTH, LENGTHKM, and HYDROSEQ
+  #' 
+  #' @examples pair_nhd_catchments(prms_line = p1_reaches_sf[p1_reaches_sf$subsegid == "3_1",],
+  #'                               prms_hrus = p1_catchments_sf,
+  #'                               min_area_overlap = 0.7,
+  #'                               xwalk_table = p2_prms_nhdv2_xwalk,
+  #'                               nhd_lines = p1_nhdv2reaches_sf)
+  #'                               
+  
+  # Identify PRMS segments that require special handling
+  drb_segs_spatial <- c("31_1","135_1","233_1","236_1","244_1","249_1","332_1","354_1","355_1","358_1","396_1","593_1","1256_1","2137_1",
```

Comment:
this could be declared in _targets.R and as an argument to the function. I don't think we'll need to change this list, but it could be helpful to have as an option

Reply:
OK, that sounds good. I placed it within the function since that vector shouldn't change, but an advantage of moving to the _targets.R file is that it's more visible.